router-tests/prometheus_improved_test.go

```diff
@@ -0,0 +1,207 @@
+package integration
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"go.opentelemetry.io/otel/sdk/metric"
+)
+
+// This file exists to be smaller than prometheus_test.go, and use better testing practices.
+// Editors like Cursor, VSCode Copilot Agent, Zed, etc have a lot of trouble with the other file due
+// to its size and testing methodology. In the new file, you should use new helpers where possible, and avoid
+// asserting the values of unrelated metrics/labels.
+
+const (
+	SchemaFieldUsageMetricName = "router_graphql_schema_field_usage_total"
+
+	WgOperationSha256  = "wg_operation_sha256"
+	WgGraphQLFieldName = "wg_graphql_field_name"
+	WgGraphQLFieldType = "wg_graphql_field_type"
+	WgOperationName    = "wg_operation_name"
+	WgOperationType    = "wg_operation_type"
```

Comment:
You can reference to the original one and use a helper function to make it snake case. In that case, we don't need to maintain two places.

Reply:
Done